client/main.go

```diff
@@ -467,6 +475,63 @@ func getNamespaceInfo(resourcePath, OSDFDirectorUrl string, isPut bool) (ns name
 	}
 }
 
+func getUrlMetadata(remoteUrl *url.URL, scheme string) (metadata UrlMetadata, err error) {
```

Comment:
Make this into a NewPelicanURL method.

(Though might not be exported?  Depending on the interactions, maybe newPelicanURL)